### PR TITLE
ConstraintTemplate versions use apiextensionsv1

### PR DESCRIPTION
**Pull request by julianKatz**
Previously, I left ConstraintTemplate (CT) versions v1alpha1 and v1beta1
using apiextensionsv1beta1.JSONSchemaProps.  I thought this appropriate,
given that was what they were previously implemented as.  v1 CT, on the
other hand, would use v1beta1 JSONSchemaProps.

This turns out to be a bad idea.

Any logic we write that interacts with OpenAPIV3Schema in a CT's
Validation section is forced to interact with OpenAPIV3Schema's type.
Before this PR, that type is apiextensionsv1beta1.JSONSchemaProps.  This
makes our life difficult, as v1 CT's version of this variable will have
a different type.  This would then require us to re-implement any logic
for this section to implement a second type, or to do some gross
conversions.

Further, apiextensionsv1beta1 will be removed from
k8s.io/apiextensions-apiserver, preventing us from upgrading that
library without doing this work.

Basically, this change is unavoidable and appropriate.

This blocks open-policy-agent/frameworks#121

Contributes to open-policy-agent/gatekeeper#550

Signed-off-by: juliankatz <juliankatz@google.com>

### CODE REVIEW

**Review by maxsmythe — APPROVED** [review]
you ran `make generate` and there were no manifest changes?

If so, LGTM

@shomron @ritazh @sozercan LGTY?

**Review by ritazh — APPROVED** [review]

**Comment by ritazh** [comment]
Getting an error running `make manifests` it could be my test env.

**Comment by julianKatz** [comment]
> `make manifests` generated diff in the following files but they are all line break updates.
> modified: config/crds/templates.gatekeeper.sh_constrainttemplates.yaml
> modified: deploy/crds.yaml

@ritazh What version of `controller-gen` are you running?  I just tried again and didn't see any diff.  Mine is:

```
❯ controller-gen --version
Version: v0.5.0
```

**Comment by ritazh** [comment]
@julianKatz it's showing `Version: v0.5.0`
